app/domain/invoices/abacus/sales_order.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+module Invoices
+  module Abacus
+    class SalesOrder < Entity
+
+      SOURCE_SYSTEM = 'hitobito'
+      BACKLOG_ID = 0
+      TYPE = 'Product'
+      TYPE_OF_PRINTING = 'AccToSequentialControl'
+      INVOICE_KINDS = {
+        membership: 'R',
+        course: 'C'
+      }.with_indifferent_access.freeze
+
+      def create(positions, additional_user_fields: {})
+        create_sales_order(additional_user_fields: additional_user_fields)
+        create_sales_order_positions(positions)
+        trigger_sales_order
+      rescue => e
+        delete_sales_order
+        raise e
+      end
+
+      def fetch
+        client.get(:sales_order, abacus_id, '$expand' => 'Positions')
+      end
+
+      private
+
+      def create_sales_order(additional_user_fields: {})
+        attrs = sales_order_attrs(additional_user_fields: additional_user_fields)
+        data = client.create(:sales_order, attrs)
+        entity.abacus_sales_order_key = data.fetch(:sales_order_id)
+      end
+
+      def create_sales_order_positions(positions)
+        positions.each_with_index do |position, index|
+          client.create(:sales_order_position, sales_order_position_attrs(position, index + 1))
+        end
+      end
+
+      def trigger_sales_order
+        path = "#{client.endpoint(:sales_order, abacus_id)}/" \
```

Comment:
Könnten wir später dann mal noch in den Client verschieben

Reply:
Mal schauen. Funktioniert eh noch nicht in Abacus...